GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRMouseDeviceManager.java

```diff
@@ -75,9 +80,17 @@ void removeCursorController(GVRBaseController controller) {
             thread.quitSafely();
             thread = new EventHandlerThread(THREAD_NAME);
             threadStarted = false;
+            if(gvrContext != null){
+                gvrContext.unregisterDrawFrameListener(this);
+            }
         }
     }
 
+    @Override
+    public void onDrawFrame(float frameTime) {
+        thread.invalidate();
```

Comment:
Or maybe I am not reading it right. There is GVRCursorController.invalidate and there is now one more (which maybe should have been called something like sendInvalidateMessage).